components/feature/night-mode/src/main/java/org/mozilla/rocket/nightmode/themed/NightTheme.kt

```diff
@@ -0,0 +1,28 @@
+package org.mozilla.rocket.nightmode.themed
+
+interface NightTheme : DefaultTheme {
+
+    fun setNightMode(isNight: Boolean) {
+        if (isNightTheme() != isNight) {
+            if (isNight) {
+                addThemeState(ThemedWidgetUtils.ThemeState.NIGHT)
+            } else {
+                removeThemeState(ThemedWidgetUtils.ThemeState.NIGHT)
+            }
+            notifyRefreshDrawableState()
+        }
+    }
+
+    fun isNightTheme(): Boolean {
+        return themeState and ThemedWidgetUtils.ThemeState.NIGHT.flag != 0
+    }
+
+    override fun getThemeStateSize(): Int {
+        return ThemedWidgetUtils.STATE_NIGHT_MODE.size
+    }
+
+    override fun getThemeDrawableState(): IntArray {
```

Comment:
Just asking: So, for NightPrivateTheme, it will implement both NightTheme and PrivateTheme and have a static intArray member which merges ThemedWidgetUtils.STATE_NIGHT_MODE and hemedWidgetUtils.STATE_PRIVATE_MODE?

Reply:
Yes, it should be like this.
`override fun getThemeDrawableState(): IntArray {
        return when {
            isNightTheme() && isPrivateTheme() -> super<NightTheme>.getThemeDrawableState().plus(super<PrivateTheme>.getThemeDrawableState())
            isNightTheme() -> super<NightTheme>.getThemeDrawableState()
            isPrivateTheme() -> super<PrivateTheme>.getThemeDrawableState()
            else -> intArrayOf(0)
        }
    }`

Reply:
Sounds good to me, ready to go.